felix/rules/dispatch.go

```diff
@@ -519,6 +526,17 @@ func (r *DefaultRuleRenderer) buildSingleDispatchChainTree(
 			nextChar := prefix[len(commonPrefix):]
 			ifaceMatch := prefix + r.wildcard
 			childChainName := chainName + "-" + nextChar
+			// Extend the chain name with the index number if greater than zero.
+			// This fixes the chain naming collision that can occur when chains
+			// are constructed in more than a single pass. Without this,
+			// interfaces like "caliX..." (Calico workload interfaces) can end
+			// up in the same chain than interfaces that start with the same
+			// character "X.." if they share the same prefix. In this case rules
+			// generated in the previous round could be overwritten by the new
+			// ones.
+			if indexNumber > 0 {
+				childChainName += fmt.Sprintf("%d", indexNumber)
```

Comment:
I think this can still hit an issue if the conflicting prefix is of the form `nextChar` followed by `index` ? 

Probably make sense to put the index indicator before the hyphen to avoid that:

```
fmt.Sprintf("%s%s-%s", chainName, index, nextChar)
```